internal/pkg/unit/unit.go

```diff
@@ -33,6 +33,30 @@ func Run(testfiles, configfiles []string) ([]string, []string, error) {
 				details = append(details, fmt.Sprintf("FAIL input:[%v]", input))
 				return summary, details, fmt.Errorf("error getting destinations: %v", err)
 			}
+			if route.Delegate != nil {
+				// If delegate is defined in the test case, assume the delegated virtual service is not in the parsed VirtualServices list.
+				// Compare the delegate configuration and skip the rest of the checks.
+				if testCase.Delegate != nil {
+					if reflect.DeepEqual(route.Delegate, testCase.Delegate) != testCase.WantMatch {
+						details = append(details, fmt.Sprintf("FAIL input:[%v]", input))
+						return summary, details, fmt.Errorf("delegate missmatch=%v, want %v, rule matched: %v", route.Delegate, testCase.Delegate, route.Match)
+					}
+					details = append(details, fmt.Sprintf("PASS input:[%v]", input))
+					continue
+				} else {
+					// Else, lookup delegated virtual service and run the rest of the checks as usual.
+					vs, err := GetDelegatedVirtualService(route.Delegate, parsed.VirtualServices)
+					if err != nil {
+						details = append(details, fmt.Sprintf("FAIL input:[%v]", input))
+						return summary, details, fmt.Errorf("error getting delegate virtual service: %v", err)
+					}
+					route, err = GetRoute(input, []*v1alpha3.VirtualService{vs})
+					if err != nil {
+						details = append(details, fmt.Sprintf("FAIL input:[%v]", input))
+						return summary, details, fmt.Errorf("error getting destinations: %v", err)
+					}
+				}
+			}
 			if reflect.DeepEqual(route.Route, testCase.Route) != testCase.WantMatch {
```

Comment:
@cainelli We could also check if `testCase.Route != nil` here instead of returning early when Delegate is set.

Reply:
Yes, better to fail with a concrete error if delegate is set and any other match type is set. (route, header, rewrite, etc).

Reply:
I changed the logic in https://github.com/getyourguide/istio-config-validator/pull/389/commits/ec0f5f3a24c02fba27578b0a0b32b6523488b7ba.

This allows the user to both test the delegate taken as well as the resulting route (see the additional test I added).

Let me know what you think.

Reply:
I was thinking in checking all match types. If delegate match is present it does not make sense to continue asserting not only route but header, rewrites, fault injection, etc.

I can follow-up when I'm back from holidays :)

Reply:
I see. Feel free to tag me in an issue / pr if you want to revisit this.

Enjoy the holiday!